share/smack/top.py

```diff
@@ -899,85 +978,145 @@ def verify_bpl(args):
+def thread_verify_bpl(args, args_to_add):
 
-        if args.error_file:
-            with open(args.error_file, 'w') as f:
-                f.write(error)
+    if 'verifier' in args_to_add:
+        if args_to_add['verifier'] is 'portfolio':
+            raise RuntimeError("portfolio is not a valid verifier"
+                               " specification within the portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument verifier from"
+                  " the chosen portfolio configuration")
+            args.verifier = args_to_add['verifier']
+    else:
+        raise RuntimeError("verifier is a required argument"
+                           " in the portfolio configuration file")
 
-        if not args.quiet:
-            print(error)
+    if 'modular' in args_to_add:
+        if args.modular:
+            raise RuntimeError("argument modular specified in both command"
+                               " line and portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument modular from"
+                  " the chosen portfolio configuration")
+            args.modular = args_to_add['modular']
+
+    if (args.verifier is not 'boogie') and args.modular:
+        raise RuntimeError("Incompatible arguments modular"
+                           " and non-boogie verifier were specified")
+
+    if 'verifier-options' in args_to_add:
+        if args.verifier_options is not '':
+            raise RuntimeError("argument verifier-options specified in both"
+                               " command line and portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument verifier-options from"
+                  " the chosen portfolio configuration")
+            args.verifier_options = args_to_add['verifier-options']
+
+    if 'solver' in args_to_add:
+        if args.solver is not 'z3': # is there a better check than not default?
+            raise RuntimeError("argument solver specified in both command"
+                               " line and portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument solver from"
+                  " the chosen portfolio configuration")
+            args.solver = args_to_add['solver']
 
-        if args.replay:
-            replay_error_trace(verifier_output, args)
-    print(result.message(args))
-    return result.return_code()
+    if args.verifier == 'boogie' or args.modular:
+        command = invoke_boogie(args)
+
+    elif args.verifier == 'corral':
+        command = invoke_corral(args)
+
+    elif args.verifier == 'symbooglix':
+        command = invoke_symbooglix(args)
+
+    if args.verifier_options:
+        command += args.verifier_options.split()
+
+    verifier_output = try_command(command, timeout=args.time_limit)
+    return args, verifier_output
+
+
+def verify_bpl_portfolio(args):
+    if args.portfolio_config is None:
+        portfolio_config = {'thread1':
```

Comment:
I actually think now we should be reading this in from a file. Call it `default-portfolio.yaml` and put it in the same folder where this script is.
Note that you have to add the pattern "*.yaml" here: https://github.com/smackers/smack/blob/main/CMakeLists.txt#L222